source/extensions/filters/http/jwt_authn/filter_config.cc

```diff
@@ -22,6 +23,21 @@ FilterConfigImpl::FilterConfigImpl(
 
   jwks_cache_ = JwksCache::create(proto_config_, context, Common::JwksFetcher::create, stats_);
 
+  // Validate provider URIs.
+  // Note that the PGV well-known regex for URI is not implemented in C++, otherwise we could add a
+  // PGV rule instead of doing this check manually.
+  for (const auto& provider_pair : proto_config_.providers()) {
```

Comment:
Let's add this with a runtime-guard, to ensure users can revert for now if the validation is too strict.